### DISCUSSION TITLE
image.AddProfile to add ExifProfile

### DISCUSSION THREAD

**Discussion by AlquistArj** [dlemstra/Magick.NET#835] [Help] [answered]
Is there any way to add an ExifProfile to a MagickImage? I found an old reply from years ago that suggested image.AddProfile can be used but it does not seem to be available. I'm trying to add an Exif profile to a .jpg that does not have one.

**Comment by dlemstra** [comment]
The method has been renamed to `SetProfile`.

**Comment by AlquistArj** [comment]
Ok, thanks. I'm doing this but not seeing a change in the file properties (when viewed from Windows 10 File Properties). Am I doing something wrong?

var profile = new ExifProfile();
profile.SetValue(ExifTag.XResolution, new Rational(300));
profile.SetValue(ExifTag.YResolution, new Rational(300));
profile.SetValue(ExifTag.ResolutionUnit, (ushort)2);
image.SetProfile(profile);
image.Write(pfn);

**Comment by dlemstra** [comment, marked as answer]
This all depends on the format of your image. It is possible that your image format use the `Density` property of the `image`. What happens when you change that instead?

**Reply by AlquistArj** [reply]
I'm attempting to add an EXIF profile to a .jpg that didn't originally have one. You are right changing image.Density successfully sets the resolution in a way that Windows File Explorer is happy to display. Thanks for your help.
